util/parseRPMrepo.py

```diff
@@ -1,80 +1,86 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # This script parses RPM repos
 
+import hashlib
 import os
 import sys
-import hashlib
-
 from xml.etree.ElementTree import ElementTree
-from xml.etree.ElementTree import dump 
 
 XMLREPONS = "{http://linux.duke.edu/metadata/repo}"
 XMLPKGNS = "{http://linux.duke.edu/metadata/common}"
 
+
 def check_file(filepath, checksumtype, checksum):
     filechecksum = ""
     if checksumtype == "sha":
-        filechecksum = hashlib.sha1(''.join(open(filepath).readlines())).hexdigest()
+        filechecksum = hashlib.sha1(b''.join(open(filepath, mode='rb').readlines())).hexdigest()
     elif checksumtype == "sha256":
-        filechecksum = hashlib.sha256(''.join(open(filepath).readlines())).hexdigest()
+        filechecksum = hashlib.sha256(b''.join(open(filepath, mode='rb').readlines())).hexdigest()
 
     if filechecksum == checksum:
         return True
     return False
 
+
 def parse_repomd(repo):
     filename = os.path.join(repo, "repodata", "repomd.xml")
     toret = {}
 
     tree = ElementTree()
     tree.parse(open(filename, 'r'))
 
-    for data in tree.findall(XMLREPONS+"data"):
+    for data in tree.findall(XMLREPONS + "data"):
```

Comment:
Better way to string concatenation is format strings like f"{XMLREPONS}data} 
Same for other similar cases.